src/transformers/models/gptsan_japanese/modeling_gptsan_japanese.py

```diff
@@ -0,0 +1,1386 @@
+                ],
+            )
+            pasts_or_spout_value = torch.split(pasts_or_spout_value, [1] * self.config.num_layers, dim=1)
+            # make same shape as past_key_values
+            pasts_or_spout_value = tuple(
+                tuple([b.squeeze(1) for b in torch.split(a.squeeze(1), [1, 1], dim=1)]) for a in pasts_or_spout_value
+            )
+        else:
+            pasts_or_spout_value = [None] * self.config.num_layers
+
+        # Token position considering spout and pasts
+        token_position = torch.arange(num_input_contexts).to(device) + num_pasts_contexts
+
+        if attention_mask is None:
+            attention_mask = torch.ones(num_batch, num_input_contexts, device=device)
+
+        # positions for get position_embeddings
+        gather_position = (
+            (
+                torch.zeros((num_batch, self.config.d_model, num_input_contexts)).to(device)
+                + token_position.unsqueeze(0)
+            )
+            .transpose(1, 2)
+            .long()
+        )
+        # When padding with padding_side="left", zeros line up on the left side of attention_mask, so position_embeddings is shifted accordingly
+        gather_position -= (1 - attention_mask).argmin(dim=-1).unsqueeze(1).unsqueeze(2)
+        gather_position = torch.clip(gather_position, num_pasts_contexts, self.config.max_position_embeddings - 1)
+
+        # attention_mask is applied per batch
+        for i in range(num_batch):
+            hidden_states[i] += torch.gather(self.position_embeddings.weight, dim=0, index=gather_position[i])
+
+        # Create a mask to be used when making the prefix Input length of Prefix-LM variable
+        # When token_type_ids=None or all zero, it is equivalent to regular causal mask
+        #
+        # ex;
+        # >>> x_token = tokenizer("ｱｲｳｴ")
+        # input_ids:      | SOT | SEG | ｱ | ｲ | ｳ | ｴ |
+        # token_type_ids: | 1   | 0   | 0 | 0 | 0 | 0 |
+        # prefix_lm_mask:
+        # SOT | 1 0 0 0 0 0 |
+        # SEG | 1 1 0 0 0 0 |
+        # ｱ   | 1 1 1 0 0 0 |
+        # ｲ   | 1 1 1 1 0 0 |
+        # ｳ   | 1 1 1 1 1 0 |
+        # ｴ   | 1 1 1 1 1 1 |
+        #
+        # >>> x_token = tokenizer("", prefix_text="ｱｲｳｴ")
+        # input_ids:      | SOT | ｱ | ｲ | ｳ | ｴ | SEG |
+        # token_type_ids: | 1   | 1 | 1 | 1 | 1 | 0  |
+        # prefix_lm_mask:
+        # SOT | 1 1 1 1 1 0 |
+        # ｱ   | 1 1 1 1 1 0 |
+        # ｲ   | 1 1 1 1 1 0 |
+        # ｳ   | 1 1 1 1 1 0 |
+        # ｴ   | 1 1 1 1 1 0 |
+        # SEG | 1 1 1 1 1 1 |
+        #
+        # >>> x_token = tokenizer("ｳｴ", prefix_text="ｱｲ")
+        # input_ids:      | SOT | ｱ | ｲ | SEG | ｳ | ｴ |
+        # token_type_ids: | 1   | 1 | 1 | 0   | 0 | 0 |
+        # prefix_lm_mask:
+        # SOT | 1 1 1 0 0 0 |
+        # ｱ   | 1 1 1 0 0 0 |
+        # ｲ   | 1 1 1 0 0 0 |
+        # SEG | 1 1 1 1 0 0 |
+        # ｳ   | 1 1 1 1 1 0 |
+        # ｴ   | 1 1 1 1 1 1 |
```

Comment:
this can be removed or put in a `Tips:` section in the model's readme referencing to this line of code!